exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/NoopSpanExporter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.trace;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.util.Collection;
+import javax.annotation.Nonnull;
+
+/**
+ * A noop implementation of a {@link SpanExporter}. This class is implemented as a singleton and
+ * return a successful result on all public methods without performing any action.
+ */
+class NoopSpanExporter implements SpanExporter {
+
+  NoopSpanExporter() {
+    // prevent explicit public call to default constructor
+  }
+
+  @Override
+  public CompletableResultCode export(@Nonnull Collection<SpanData> spans) {
+    return CompletableResultCode.ofFailure();
```

Comment:
The rationale behind this is that a NoopSpanExporter (a no-op implementation of TraceExporter) can be expected to `flush` or `shutdown`, but is not able to `export` - hence returning failure.

Reply:
This should succeed.  The only reason here is if you have hope of recovery or you need the upstream system to "stop".

Unfortunately, given the runtime you're in, there's no good way to stop, you have no hope of recovery and you've already logged the actual error.  This should return `ofSuccess`

Reply:
Reopening this.  This should be returning a success status code.  The only thing returning failure here is likely to do would be cause log-spam (at best) or down the SDK completely (at worst), providing no additional value to the user over the initial failure log statement that was dumped.